### PR TITLE
Feature/ruby server

### PR DESCRIPTION
**Pull request by mikeshaw-stripe**
Adding the backend for Ruby

### CODE REVIEW

File: server/ruby/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+Copy the example environment variables file `.env.example` from the root of the repo into your own environment file called `.env`:
+
+```
+cp .env.example .env
+```
+
+User `bundler` to install the required gems
+
```
Comment (thorsten-stripe) [inline server/ruby/README.md]: maybe add that one should navigate to `server/ruby` before running these commands? Should be clear, but still.

File: server/ruby/inventory.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+require 'stripe'
+require 'dotenv'
+
+Dotenv.load(File.dirname(__FILE__) + '/../../.env')
+
+Stripe.api_key = ENV['STRIPE_SECRET_KEY']
+Stripe.api_version = '2019-02-11'
+
+# For product retrieval and listing set API version to 2018-02-28 so that skus are returned.
+@@product_api_stripe_version = '2018-02-28'
```
Comment (thorsten-stripe) [inline server/ruby/inventory.rb]: running it I get. Not sure if that's bad
```$ bundle exec ruby app.rb
/Users/thorsten/Documents/stripe-payments-demo/server/ruby/inventory.rb:10: warning: class variable access from toplevel```

File: server/ruby/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+### Testing Webhooks
+
+If you want to test [receiving webhooks](https://stripe.com/docs/webhooks), we recommend using ngrok to expose your local server.
+
+First [download ngrok](https://ngrok.com) and start your Sinatra application.
+
+[Run ngrok](https://ngrok.com/docs). Assuming your Sinatra application is running on the default port 4567, you can simply run ngrok in your Terminal in the directory where you downloaded ngrok:
+
+```
+./ngrok http 4567
```
Comment (thorsten-stripe) [inline server/ruby/README.md]: do you have to do `./`? for me ngrok is installed globally so I can do `ngrok http 4567`

File: server/ruby/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+ngrok will display a UI in your terminal telling you the new forwarding address for your Sinatra app. Use this URL as the URL to be called in your developer [webhooks panel.](https://dashboard.stripe.com/account/webhooks)
+
+Don't forget to append `/webhook` when you set up your Stripe webhook URL in the Dashboard. Example URL to be called: `http://75795038.ngrok.io/webhook`.
```
Comment (thorsten-stripe) [inline server/ruby/README.md]: we should default to https usage

File: server/ruby/app.rb [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    paymentIntent: payment_intent
+  }.to_json
+end
+
+post '/payment_intents/:id/shipping_change' do
+  content_type 'application/json'
+  data = JSON.parse request.body.read
+
+  amount = Inventory.calculate_payment_amount(data['items'])
+  amount += Inventory.get_shipping_cost(data['shippingOption']['id'])
```
Comment (thorsten-stripe) [inline server/ruby/app.rb]: Throws error `nil can't be coerced into Integer`
image

Comment (thorsten-stripe) [reply server/ruby/app.rb]: Fixed in https://github.com/stripe/stripe-payments-demo/pull/54/commits/936a1f7c762bc851c33f217852850d49ca7d4915

File: server/ruby/inventory.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+class Inventory
+  def self.calculate_payment_amount(items)
+    Stripe.api_version = @@product_api_stripe_version
```
Comment (thorsten-stripe) [inline server/ruby/inventory.rb]: We should use per request configuration instead: https://github.com/stripe/stripe-ruby#per-request-configuration

File: server/ruby/setup.rb [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+    skus = [
+      {
+        id: 'increment-03',
+        product: 'increment',
+        attributes: {
+          issue: 'Issue #3 “Development”'
+        },
+        price: 399,
+        currency: 'usd',
```
Comment (thorsten-stripe) [inline server/ruby/setup.rb]: We should add a note to the README that the currency is hardcoded in the setup script.

File: server/ruby/README.md (hunk 3 above)
Comment (mikeshaw-stripe) [reply server/ruby/README.md]: This was a C&P from the python README, but you are correct

File: server/ruby/README.md (hunk 4 above)
Comment (mikeshaw-stripe) [reply server/ruby/README.md]: Ack. We should fix in Python README as well

File: server/ruby/inventory.rb [hunk 8]
```diff
@@ -33,12 +26,17 @@ def self.get_shipping_cost(id)
   end
 
   def self.list_products
-    Stripe.api_version = @@product_api_stripe_version
     Stripe::Product.list(limit: 3)
   end
 
+  def self.list_skus(product_id)
+    Stripe::SKU.list(
+      limit: 3,
```
Comment (thorsten-stripe) [inline server/ruby/inventory.rb]: Should this be 1, seeing that we only have 1 sku per product?

**Pull request merged**